exonum-java-binding-core/rust/src/proxy/service.rs

```diff
@@ -75,15 +75,16 @@ impl Service for ServiceProxy {
     fn state_hash(&self, snapshot: &Snapshot) -> Vec<Hash> {
         unwrap_jni(self.exec.with_attached(|env| {
             let view_handle = to_handle(View::from_ref_snapshot(snapshot));
-            let java_service_hashes = panic_on_exception(
-                env,
-                env.call_method(
+            let java_service_hashes = panic_on_exception(env, unsafe {
+                env.call_method_unsafe(
                     self.service.as_obj(),
-                    "getStateHashes",
-                    "(J)[[B",
+                    jni_cache::get_usa_state_hashes(),
+                    JavaType::Array(Box::new(JavaType::Array(Box::new(JavaType::Primitive(
```

Comment:
Here and elsewhere: Why don't we add imports for this (_in this particular case,_ I'd even prefer `JavaType#from_str/parse("[[B")` to such a long and deeply nested chain)?

Reply:
Importing enum variants are usually considered bad practice. I agree that we can use `from_str` here.